src/components/Filter/Filter.tsx

```diff
@@ -0,0 +1,101 @@
+import filtersData from "@/data/filters.json";
+import { Filter as FilterType } from "@/types/Filter";
+import React, { useState } from "react";
+
+interface FilterProps {
+  onFilterChange: (selectedOptions: Record<string, string[]>) => void;
+  categoryId: number; 
+}
+
+export const Filter: React.FC<FilterProps> = ({ onFilterChange, categoryId }) => {
```

Comment:
Aquí definiste el tipo pero no lo usaste


```suggestion
export const Filter: React.FC<FilterProps> = ({ onFilterChange, categoryId }: FilterProps) => {
```